public/app/plugins/panel/annolist/AnnoListPanel.tsx

```diff
@@ -146,10 +146,10 @@ export class AnnoListPanel extends PureComponent<Props, State> {
     const params = {
       from: this._timeOffset(anno.time, options.navigateBefore, true),
       to: this._timeOffset(anno.timeEnd ?? anno.time, options.navigateAfter, false),
-      viewPanel: options.navigateToPanel ? anno.panelId : undefined,
+      viewPanel: options.navigateToPanel && anno.panelId ? anno.panelId : undefined,
```

Comment:
Without this, the URL would update with viewPanel=null, which would then display an error message.